src/components/TarsiaGrid/index.jsx

```diff
@@ -3,42 +3,31 @@ import Triangle from '../Triangle'
 
 import appConfig from '../../config'
 
-class TarsiaGrid  extends React.Component {
-  constructor(props) {
-    super(props);
-    this.id = props.id;
-    this.config = props.config
-    this.values = props.values
-  }
-
-  render() {
-    const grid = []
-    for (let tri of this.config) {
-      let valueArray = []
-      for (let valueCode of tri.values) { // lookup the value code from the config
-        if (valueCode) { // handles nulls
-          let valueCodeSplit = [valueCode.substr(0,valueCode.length-1), valueCode.slice(-1)] // split into number and q/a
-          if (this.values[valueCodeSplit[0]]) { // handles nulls
-            valueArray.push(this.values[valueCodeSplit[0]][valueCodeSplit[1]]) // add the value based on the question number and q/a
-          } else (
-            valueArray.push(valueCode)
-          )
-        } else {
-          valueArray.push(null) // null handling
-        }
+const TarsiaGrid = ({ id, config, values }) => {
```

Comment:
Moved from a react class component to a react function component: https://reactjs.org/docs/components-and-props.html

I also used the arrow function syntax. I prefer this style and it avoids the need to use the `this` keyword and binding: https://www.w3schools.com/js/js_arrow_function.asp